### ISSUE TITLE
Rebuild loop when running nuxi dev

### ISSUE THREAD

**Issue by dwightjack** [nuxt/bridge#832]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     2.17.0
- Nitro Version:    2.4.1
- Package Manager:  pnpm@8.5.0
- Builder:          webpack
- User Config:      head, components, build, serverHandlers, devServerHandlers, devServer, typescript, nitro, bridge, buildModules
- Runtime Modules:  -
- Build Modules:    (), @nuxt/bridge@3.0.0-28109745.a6e6787

### Reproduction

- Repo: https://github.com/dwightjack/demo-nuxt-bridge
- Installation: `pnpm run dev`
- Dev: `pnpm run dev`

### Describe the bug

On a fresh install of Nuxt and Nuxt Bridge, I often experience rebuild loops in development mode.

Sometimes the error is `Cannot read properties of null (reading 'close')`, more often the message is ` .nuxt/dist directory has been removed. Restarting nuxt...`. I attach a sample video of what's happening:

https://github.com/nuxt/bridge/assets/104721/10ee783b-c997-4d83-8958-a24d559e3b77

### Logs

```shell
● Client █████████████████████████ building (64%) 450/455 modules 5 active
 node_modules/.pnpm/@babel+runtime@7.22.5/node_modules/@babel/runtime/helpers/esm/superPropBase.js

✔ Server
  Compiled successfully in 2.61s
ℹ .nuxt/dist directory has been removed. Restarting nuxt...                                                                           12:42:43 PM

 FATAL  Cannot read properties of null (reading 'close')
```
```

**Comment by wattanx** [comment]
The version of bridge appears to be out of date.
Does this still occur with the latest version?
`"@nuxt/bridge": "npm:@nuxt/bridge-edge@3.0.0-28125529.9b0a98e"`

**Comment by dwightjack** [comment]
@wattanx Thanks for the reply. I updated the version, removed the lock file, and reinstalled it, but the issue is still there.

Here is another reproduction video (I also updated the reproduction repo)

https://github.com/nuxt/bridge/assets/104721/63f21e99-6a4a-4ff7-98bd-26f678125137

**Comment by dwightjack** [comment]
@wattanx An update. I tried to run the same repo on a M1 Macbook with more RAM and the issue is happening less often. The only thing that changes seems to be that on the M1 the task completes faster.

**Comment by ColinRosati** [comment]
I can reproduce this

**Comment by wattanx** [comment]
Thanks for the report!!
It appears that this is happening only with webpack.
I will investigate this in more detail.

https://stackblitz.com/edit/github-3fgb91-mgegmc

```log
ℹ .nuxt/dist directory has been removed. Restarting nuxt...                           19:26:36

 FATAL  Cannot read properties of null (reading 'close')                              19:26:36

  at Builder.unwatch (node_modules/@nuxt/builder/dist/builder.js:713:30)
  at Builder.close (node_modules/@nuxt/builder/dist/builder.js:721:10)
  at eval (node_modules/@nuxt/builder/dist/builder.js:179:42)
  at eval (node_modules/hable/dist/hable.js:1:1065)
  at eval (node_modules/hable/dist/hable.js:1:283)
  at async Nuxt.callHook (node_modules/hable/dist/hable.js:1:1034)
```

**Comment by bradley-varol** [comment]
Seeing exactly the same issue and same `.nuxt/dist directory has been removed. Restarting nuxt...` message

**Comment by wattanx** [comment]
📝 

Setting `build.quiet: true` made the logs easier to read.
```log
Nuxi 3.6.5                                                                            17:57:12
Nuxt 2.17.1 with Nitro 2.5.2                                                          17:57:12
                                                                                      17:57:12
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.1.104:3000/

ℹ Module meta took 1.5ms to setup.                                                   17:57:13
ℹ Module nuxt-bridge took 5.68ms to setup.                                           17:57:13
ℹ Module imports took 0.44ms to setup.                                               17:57:13
ℹ .nuxt/dist directory has been removed. Restarting nuxt...                          17:57:132

 FATAL  Cannot read properties of null (reading 'close')                              17:57:13

  at Builder.unwatch (node_modules/@nuxt/builder/dist/builder.js:713:30)
  at Builder.close (node_modules/@nuxt/builder/dist/builder.js:721:10)
  at eval (node_modules/@nuxt/builder/dist/builder.js:179:42)
  at eval (node_modules/hable/dist/hable.js:1:1065)
  at eval (node_modules/hable/dist/hable.js:1:283)
  at async Nuxt.callHook (node_modules/hable/dist/hable.js:1:1034)
  at async Nuxt.close (node_modules/@nuxt/core/dist/core.js:359:5)
  at async load (node_modules/nuxi/dist/chunks/dev.mjs:193:11)
  at async _applyPromised (node_modules/nuxi/dist/chunks/dev.mjs:112:10)

ℹ Module meta took 0.26ms to setup.                                                  17:57:13
ℹ Module nuxt-bridge took 0.86ms to setup.                                           17:57:13
ℹ Module imports took 0.26ms to setup.                                               17:57:13
✔ Nitro built in 261 ms                                                        nitro 17:57:16
Nuxi 3.6.5                                                                            17:57:16
                                                                                      17:57:16
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.1.104:3000/


```

**Comment by wattanx** [comment]
I confirmed that the latest nuxi solved the problem. (`nuxi-edge@3.7.0-28199397.4e409614`)
https://stackblitz.com/edit/github-3fgb91-c32oex

This will be resolved when Nuxt 3.7.0 is released!

**Comment by Rosendito** [comment]
@wattanx I love u, u have saved my life!!

**Comment by jeanpierrecarvalho** [comment]
@wattanx awesome!

**Comment by wattanx** [comment]
Resolved with version `3.0.0-28232126.5c5bc70`.

**Comment by wJoenn** [comment]
@wattanx I'm having the same issue while running
- Nuxt 2.17.2
- Bridge 3.0.0-rc.4-28381778.ba2b18e
- Nuxi  3.10.0

My dev server keeps recompiling/rebuilding indefinitely
Should I open a new issue ?
![image](https://github.com/nuxt/bridge/assets/75388869/410aa3b6-e8cb-4ac5-831a-7c15046d7b28)
